.NET/Microsoft.Recognizers.Text.DateTime/Spanish/Parsers/SpanishTimeParserConfiguration.cs

```diff
@@ -1,120 +1,121 @@
-        {
-            var trimedSuffix = suffix.Trim().ToLowerInvariant();
-            AdjustByPrefix(trimedSuffix, ref hour, ref min, ref hasMin);
-
-            var deltaHour = 0;
-            var match = SpanishTimeExtractorConfiguration.TimeSuffix.Match(trimedSuffix);
-            if (match.Success && match.Index == 0 && match.Length == trimedSuffix.Length)
-            {
-                var oclockStr = match.Groups["oclock"].Value;
-                if (string.IsNullOrEmpty(oclockStr))
-                {
-                    var amStr = match.Groups["am"].Value;
-                    if (!string.IsNullOrEmpty(amStr))
-                    {
-                        if (hour >= 12)
-                        {
-                            deltaHour = -12;
-                        }
-                        hasAm = true;
-                    }
-
-                    var pmStr = match.Groups["pm"].Value;
-                    if (!string.IsNullOrEmpty(pmStr))
-                    {
-                        if (hour < 12)
-                        {
-                            deltaHour = 12;
-                        }
-                        hasPm = true;
-                    }
+        public SpanishTimeParserConfiguration(ICommonDateTimeParserConfiguration config)
+        {
+            TimeTokenPrefix = DateTimeDefinitions.TimeTokenPrefix;
+            AtRegex = SpanishTimeExtractorConfiguration.AtRegex;
+            TimeRegexes = SpanishTimeExtractorConfiguration.TimeRegexList;
+            UtilityConfiguration = config.UtilityConfiguration;
+            Numbers = config.Numbers;
+        }
+
+        public void AdjustByPrefix(string prefix, ref int hour, ref int min, ref bool hasMin)
+        {
+            var deltaMin = 0;
+            var trimedPrefix = prefix.Trim().ToLowerInvariant();
+
+            if (trimedPrefix.StartsWith("cuarto") || trimedPrefix.StartsWith("y cuarto"))
+            {
+                deltaMin = 15;
+            }
+            else if (trimedPrefix.StartsWith("menos cuarto"))
+            {
+                deltaMin = -15;
+            }
+            else if (trimedPrefix.StartsWith("media") || trimedPrefix.StartsWith("y media"))
+            {
+                deltaMin = 30;
+            }
+            else
+            {
+                var match = SpanishTimeExtractorConfiguration.LessThanOneHour.Match(trimedPrefix);
+                var minStr = match.Groups["deltamin"].Value;
+                if (!string.IsNullOrWhiteSpace(minStr))
+                {
+                    deltaMin = int.Parse(minStr);
+                }
+                else
+                {
+                    minStr = match.Groups["deltaminnum"].Value.ToLower();
+                    Numbers.TryGetValue(minStr, out deltaMin);
+                }
+            }
+
+            if (trimedPrefix.EndsWith("pasadas") || trimedPrefix.EndsWith("pasados") ||

```

Comment:
These strings should also be moved to definitions, or become a regex there.

Reply:
After all definitions extraction are merged, we'll create a new PR (for all languages) with these changes